xbmc/video/VideoDimensions.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+/*
+ *      Copyright (C) 2005-2013 Team XBMC
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <string>
+#include <map>
+#include <vector>
+#include <utility>
+
+typedef std::pair<std::pair<int, int>, std::pair<int, int>> Boundary;
+
+class CVideoDimensions
+{
+public:
+  CVideoDimensions(int width, int height);
+  ~CVideoDimensions() {};
+
+  /*
+   * @return a descriptor for the resolution, e.g. "720"
+   */
+  std::string GetDescriptor() const;
+
+  /*
+   * @return the fidelity of the resolution, e.g. "SD" or "HD"
+   */
+  std::string GetFidelity() const;
+
+  /*
+   * @return a list of all descriptors
+   */
+  static std::vector<std::string> GetDescriptors();
+
+  /*
+   * @param descriptor a resolution descriptor, e.g. "720"
+   * @return the dimensional boundaries for the specified descriptor
+   * @see CVideoDimensions::boundaries
+   */
+  static Boundary GetBoundary(const std::string &descriptor);
+
+private:
+  int m_width;
+  int m_height;
+
+  /*
+   * Maps a resolution descriptor to a set of dimensional boundaries. The 
+   * boundary is represented as a pair of pairs, where the first pair is the 
+   * lower resp. upper bound for the width and the second pair is the lower 
+   * resp. upper bound for the height.
+   */
+  static std::map<std::string, Boundary> boundaries;
+
+  /*
+   * Maps a descriptor (e.g. "720") to a fidelity (e.g. "HD")
+   */
+  static std::map<std::string, std::string> fidelities;
```

Comment:
In a second thought, the "fidelity" is not related to a "boundary" at all, so without changing the name of `Boundary` I wouldn't want to get rid of that second map.